CleanWeather/Database/CityDatabase.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  CityDatabase.swift
+//  CleanWeather
+//
+//  Created by Marek Skrzelowski on 05/12/2019.
+//  Copyright © 2019 vandermesis. All rights reserved.
+//
+
+import Foundation
+
+protocol CityDatabase {
+    func saveFavouriteCity(city: FavouriteCity)
+    func loadFavouriteCities() -> [FavouriteCity]
+}
+
+final class CityDatabaseImpl: CityDatabase {
+
+    var favouriteCities = [FavouriteCity]()
+
+    let dataFilePath = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first?.appendingPathComponent("Items.plist")
+
+    func saveFavouriteCity(city: FavouriteCity) {
+
+        if !favouriteCities.contains(where: { $0.name == city.name }) {
+            favouriteCities.append(city)
+        } else {
+            for record in favouriteCities where record.name == city.name {
+                print("Unlike city")
+                //FIXME: Why I can't assign to property: record is let constant?
+//                record.favourite = false
```

Comment:
nie mozesz przypisac bo record jest wynikiem pętli for each, i nie jest to mutowalne


zapis musi polegac na odczycie wszystkiego, modyfikacji i zapisie

jak przerobisz na user defaultsy, to nie trzymaj tych favouriceCities w property

zrob tak:

- publiczne 3 metody: getFavouriteCities, addFavouriteCity, removeFavouriteCity
- prywatna: saveFavouriteCities(cities)

w add i remove bedziesz brał GETem, modyfikowal - appendujac, lub removujac z array - i robil SAVE z nową listą